common/sensor/sensor.c

```diff
@@ -1,249 +1,349 @@
+#include "sensor.h"
+
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
+
+#include "power_status.h"
 #include "sdr.h"
-#include "pal.h"
-#include "sensor.h"
-#include "sensor_def.h"
+#include "hal_i2c.h"
+#include "plat_sensor_table.h"
+#include "plat_sdr_table.h"
+#include "adm1278.h"
+#include "isl69260.h"
+#include "adc.h"
+#include "nvme.h"
+#include "pch.h"
+#include "peci.h"
+#include "tmp75.h"
+
+#define RETRY 5
 
 struct k_thread sensor_poll;
-K_KERNEL_STACK_MEMBER(sensor_poll_stack, sensor_poll_stack_size);
+K_KERNEL_STACK_MEMBER(sensor_poll_stack, SENSOR_POLL_STACK_SIZE);
 
-uint8_t SnrNum_SnrCfg_map[SENSOR_NUM_MAX];
-uint8_t SnrNum_SDR_map[SENSOR_NUM_MAX];
+uint8_t sensor_config_index_map[SENSOR_NUM_MAX];
+uint8_t sdr_index_map[SENSOR_NUM_MAX];
 
-bool enable_sensor_poll = 1;
-static bool snr_poll_eanble_flag = 1;
+bool enable_sensor_poll_thread = true;
+static bool sensor_poll_enable_flag = true;
 
-const int negative_ten_power[16] = {1,1,1,1,1,1,1,1000000000,100000000,10000000,1000000,100000,10000,1000,100,10};
+const int negative_ten_power[16] = { 1,	    1,		1,	   1,	     1,	      1,
+				     1,	    1000000000, 100000000, 10000000, 1000000, 100000,
+				     10000, 1000,	100,	   10 };
 
 snr_cfg *sensor_config;
+uint8_t sensor_config_num;
+
+static void init_sensor_num(void)
+{
+	for (int i = 0; i < SENSOR_NUM_MAX; i++) {
+		sdr_index_map[i] = 0xFF;
+		sensor_config_index_map[i] = 0xFF;
+	}
+}
+
+void map_sensor_num_to_sdr_cfg(void)
+{
+	uint8_t i, j;
+
+	for (i = 0; i < SENSOR_NUM_MAX; i++) {
+		for (j = 0; j < SDR_NUM; j++) {
+			if (i == full_sdr_table[j].sensor_num) {
+				sdr_index_map[i] = j;
+				break;
+			} else if (i == SDR_NUM) {
+				sdr_index_map[i] = SENSOR_NULL;
+			}
+		}
+		for (j = 0; j < SDR_NUM; j++) {
+			if (i == sensor_config[j].num) {
+				sensor_config_index_map[i] = j;
+				break;
+			} else if (i == SDR_NUM) {
+				sensor_config_index_map[i] = SENSOR_NULL;
+			}
+		}
+	}
+	return;
+}
+
+bool access_check(uint8_t sensor_num)
+{
+	bool (*access_checker)(uint8_t);
+
+	access_checker = sensor_config[sensor_config_index_map[sensor_num]].access_checker;
+	return (access_checker)(sensor_config[sensor_config_index_map[sensor_num]].num);
+}
+
+void clear_unaccessible_sensor_cache()
+{
+	uint8_t poll_num;
+
+	for (poll_num = 0; poll_num < SENSOR_NUM_MAX; poll_num++) {
+		if (sensor_config_index_map[poll_num] == SENSOR_NULL) { // sensor not exist
+			continue;
+		}
+
+		if (!access_check(poll_num)) {
+			sensor_config[sensor_config_index_map[poll_num]].cache = SENSOR_FAIL;
+			sensor_config[sensor_config_index_map[poll_num]].cache_status =
+				SENSOR_INIT_STATUS;
+		}
+	}
+}
+
+bool sensor_read(uint8_t sensor_num, float *reading)
```

Comment:
check if reading is NULL

Reply:
We have fixed it and done the forced update.
Please review it. Thanks.